test/test_function_schema.py

```diff
@@ -112,6 +112,53 @@ def test_string_optional_parameter_default_value(self):
         schema_b = parse_schema(str(schema_a))
         self.assertEquals(schema_a, schema_b)
 
+    def test_forward_compatible_arguments_without_out(self):
+        old_schema = parse_schema('any(Tensor self, int a, int b=1) -> Tensor')
+        new_schema = parse_schema('any(Tensor self, int a) -> Tensor')
+        is_fc, _ = new_schema.check_forward_compatible_with(old_schema)
+        self.assertTrue(is_fc)
+        new_schema = parse_schema('any(Tensor self, int a, int b=1, int c=1) -> Tensor')
+        is_fc, _ = new_schema.check_forward_compatible_with(old_schema)
+        self.assertTrue(is_fc)
+        new_schema = parse_schema('any(Tensor self, int a, int b=1, int[2] c=1) -> Tensor')
+        is_fc, reason = new_schema.check_forward_compatible_with(old_schema)
+        self.assertFalse(is_fc)
+        self.assertEqual(reason, "Function schema is not forward compatible since the new argument"
+                                 " \'c\' of type int[] has a container type as its\' default value.")
+        new_schema = parse_schema('any(Tensor self, int a, int b=4) -> Tensor')
+        is_fc, reason = new_schema.check_forward_compatible_with(old_schema)
+        self.assertFalse(is_fc)
+        self.assertEqual(reason, "\'b\' is not forward compatible with the older version of the schema")
+        new_schema = parse_schema('any(Tensor self, int a, int c=1) -> Tensor')
+        is_fc, reason = new_schema.check_forward_compatible_with(old_schema)
+        self.assertFalse(is_fc)
+        self.assertEqual(reason, "\'c\' is not forward compatible with the older version of the schema")
+        new_schema = parse_schema('any(Tensor self, int a, int c=1, int b=1) -> Tensor')
```

Comment:
What about a positional arg becoming kwarg-only and a kwarg-only arg becoming positional?

Reply:
good test case. I will add it.